tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,563 @@
+		if err != nil && result.Code != 0 {
+			fssh.LogResult(result)
+			return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				setPermissionForK8sVms, masterIp, err)
+		}
+	}
+	return nil
+}
+
+// setClusterLevelPermission is used to set cluster level permissions for test user
+func setClusterLevelPermission(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, testUserAlias string, testUser string, cluster string) error {
+	setPermissionForCluster := govcLoginCmd(vcAddress, vCenterPort) + "govc permissions.set -principal " +
+		testUserAlias + " -propagate=false -role CNS-HOST-CONFIG-STORAGE-" + testUser + " " + cluster + " | tr -d '\n'"
+	framework.Logf("Assign cluster level Permissions to user: %s ", setPermissionForCluster)
+	result, err := sshExec(sshClientConfig, masterIp, setPermissionForCluster)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			setPermissionForCluster, masterIp, err)
+	}
+	return nil
+}
+
+// setDataStoreLevelPermission is used to set datastore level permissions for test user
+func setDataStoreLevelPermission(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, testUserAlias string, testUser string, dataCenter string, datastores []string) error {
+	for i := 0; i < len(datastores); i++ {
+		setPermissionForDataStore := govcLoginCmd(vcAddress, vCenterPort) + "govc permissions.set -principal " +
+			testUserAlias + " " +
+			"-propagate=false -role CNS-DATASTORE-" + testUser + " '" + datastores[i] + "'"
+		framework.Logf("Assign datastore level Permissions to user: %s ", setPermissionForDataStore)
+		result, err := sshExec(sshClientConfig, masterIp, setPermissionForDataStore)
+		if err != nil && result.Code != 0 {
+			fssh.LogResult(result)
+			return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+				setPermissionForDataStore, masterIp, err)
+		}
+	}
+	return nil
+}
+
+// setSearchlevelPermission method is used to set search level permissions
+func setSearchlevelPermission(vcAddress string, vCenterPort string, sshClientConfig *ssh.ClientConfig,
+	masterIp string, testUserAlias string, testUser string) error {
+	setSearchLevelPermission := govcLoginCmd(vcAddress, vCenterPort) + "govc permissions.set -principal " + testUserAlias +
+		" -propagate=false -role CNS-SEARCH-AND-SPBM-" + testUser + " /"
+	framework.Logf("Assign search level Permissions to user: %s ", setSearchLevelPermission)
+	result, err := sshExec(sshClientConfig, masterIp, setSearchLevelPermission)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			setSearchLevelPermission, masterIp, err)
+	}
+	return nil
+}
+
+// createCsiVsphereSecret method is used to create csi vsphere secret file
```

Comment:
I think there are few more things which need to be added from snapshots perspective in the conf file